cmd/podman/machine/platform.go

```diff
@@ -4,10 +4,28 @@
 package machine
 
 import (
+	"os"
+	"strings"
+
 	"github.com/containers/podman/v4/pkg/machine"
 	"github.com/containers/podman/v4/pkg/machine/qemu"
+	"github.com/sirupsen/logrus"
 )
 
-func GetSystemDefaultProvider() machine.VirtProvider {
-	return qemu.GetVirtualizationProvider()
+func GetSystemProvider(name string) machine.VirtProvider {
+	provider := name
+	providerOverride, found := os.LookupEnv("CONTAINERS_MACHINE_PROVIDER")
+	if found {
+		provider = providerOverride
+	}
+	provider = strings.ToUpper(strings.TrimSpace(provider))
+	switch provider {
+	case "QEMU":
+		return qemu.GetVirtualizationProvider()
+	default:
+		if len(provider) > 0 {
+			logrus.Warnf("Unuspported provider specified %q. Will use default provider.", provider)
+		}
+		return qemu.GetVirtualizationProvider()
```

Comment:
This should return a hard error. Invalid input should be rejected, if I have a typo I want to know it and not that podman just decides to to something different.

Reply:
> This should return a hard error.

This was a tough choice for an external contributor as it there are no guidelines and no opportunity to discuss in advance (probably I should try Discord first). Can be changed according to requirements.

Reply:
Now it returns error for all sort of possible failures